### PR TITLE
New module: krakentools/kreport2krona

### LINKED ISSUE #1741: new module: KRAKENTOOLS/KREPORT2KRONA
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by MillironX**
This module adds support for [KrakenTools](https://github.com/jenniferlu717/KrakenTools)' `kreport2krona` utility.

## PR checklist

Closes #1741 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/krakentools/kreport2krona/main.nf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    kreport2krona.py \\
+        -r ${kreport} \\
+        -o ${prefix}.krona \\
```
Comment (jfy133) [inline modules/krakentools/kreport2krona/main.nf:28]: I normally prefer to specify the fileformat as an extension (and also the out channel name), which would be `.txt` in this case, but it's not so strict if you strongly prefer krona.

File: modules/krakentools/kreport2krona/main.nf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        ${args}
+
+    # Remove ugly 'x__' prefixes for each of the taxonomic levels
+    LEVELS=(d k p c o f g s)
+    for L in "\${LEVELS[@]}"; do
+        sed -i "s/\${L}__//g" ${prefix}.krona
+    done
+
+    # Remove underscores that are standing in place of spaces
+    sed -i "s/_/ /g" ${prefix}.krona
```
Comment (jfy133) [inline modules/krakentools/kreport2krona/main.nf:38]: Whule it's useful, this stuff is 'custom' not natively support by the tool, which shouldn't be in the main module (we would make a local module in the pipeline for this), so it should be removed. If you'd be willing to make the local module for taxprofiler that woud be awesome (but no worries if you don't have time)!

File: modules/krakentools/kreport2krona/meta.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - krona:
+      type: file
+      description: Krona text file converted from Kraken report
```
Comment (jfy133) [inline modules/krakentools/kreport2krona/meta.yml:37]:
```suggestion
      description: Krona text-based input file converted from Kraken report
```

File: tests/modules/krakentools/kreport2krona/test.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+- name: krakentools kreport2krona test_krakentools_kreport2krona
+  command: nextflow run ./tests/modules/krakentools/kreport2krona -entry test_krakentools_kreport2krona -c ./tests/config/nextflow.config  -c ./tests/modules/krakentools/kreport2krona/nextflow.config
+  tags:
+    - krakentools/kreport2krona
+    - krakentools
+  files:
+    - path: output/kraken2/test.kraken2.report.txt
+      md5sum: 4227755fe40478b8d7dc8634b489761e
```
Comment (jfy133) [inline tests/modules/krakentools/kreport2krona/test.yml:8]:
```suggestion
```

**Review by jfy133 — CHANGES_REQUESTED** [review]
Pretty good! Main thing is taking out the custom clean up of the contents of the file (see comment).

**Comment by MillironX** [comment]
Changes made: ready for re-review

**Review by jfy133 — APPROVED** [review]

**Comment by jfy133** [comment]
Thanks very much @MillironX !

Would you mind adding this into Taxprofiler now? And then you could re-add in a separate module the clean up thing.

**Pull request merged**